sequence_transform.c

```diff
@@ -21,10 +21,11 @@
 
 //////////////////////////////////////////////////////////////////
 // Key history buffer
-static st_key_action_t key_buffer_data[SEQUENCE_MAX_LENGTH] = {{KC_SPC, ST_DEFAULT_KEY_ACTION}};
+#define KEY_BUFFER_CAPACITY SEQUENCE_MAX_LENGTH + COMPLETION_MAX_LENGTH
```

Comment:
shouldn't generator handle such define?

Reply:
The generator is still controlling the value of, since both `SEQUENCE_MAX_LENGTH` and `COMPLETION_MAX_LENGTH` are controlled by the generator. I think this method is best, because it doesn't allow an incompaltibly small `KEY_BUFFER_CAPACITY`.

That said, we do need to talk about how we want to allow the user to add extra buffer space if they desire, because the Enhanced Backspace feature's ability to chain undos scales with the buffer size (bigger buffer means you get more recoverable history).

Reply:
I agree with Matt. Generator should only output the narrowest things it can report.
It doesn't know how we will use this stuff, and this way we don't have to update it every time we decide to do different things with these defines.
Regarding bigger buffer: this also means more rule search time when the buffer is full and there wasn't a previously found rule, since it starts from the beginning each time. If we restrict it to starting at the last space, this would be optimized, at the cost of not being able to find rules that include multiple words.

Reply:
I think that we can have two values here. One is the length of the buffer and the other is the place where you start looking for missed rules.